laser_slam_ros/include/laser_slam_ros/laser_slam_worker.hpp

```diff
@@ -43,6 +43,10 @@ class LaserSlamWorker {
   // Get a filtered map and apply map separation if desired.
   void getFilteredMap(laser_slam_ros::PointCloud* filtered_map);
 
+  // Get a vector containing the point clouds that have been recorded since
+  // the last call to this method. This call clears the point cloud queue.
+  std::vector<laser_slam_ros::PointCloud> getQueuedPoints();
```

Comment:
For vectors we always had the discussion about returning by value or using a pointer as an argument. I can't remember what was our solution for vectors containing non standard variable types. I always use the return by pointer fo vectors. Do you have a strong opinion on this?